Add the google reports drive documentation (#1266)
SEKOIA-IO/documentation: docs/xdr/features/collect/integrations/cloud_and_saas/google/google_drive_reports.md

uuid: bb11239e-3dc6-11ee-be56-0242ac120002
name: Google Drive Reports
type: intake

## Overview

Google Reports is a data reporting and analysis platform offered by Google, designed to provide insights and metrics about user activities and interactions within various Google services. It allows organizations to track and visualize user engagement, application usage, and other relevant data points, enabling informed decision-making and optimization of digital experiences. And Google Reports Drive is just for the Google Drive service.

{!_shared_content/operations_center/detection/generated/suggested_rules_bb11239e-3dc6-11ee-be56-0242ac120002_do_not_edit_manually.md!}

{!_shared_content/operations_center/integrations/generated/bb11239e-3dc6-11ee-be56-0242ac120002.md!}

## Google Cloud configuration procedure

> #### Prerequisites

Reviewer: ```suggestion
### Prerequisites
```

- Google licence Enterprise standard or higher
- Access to Sekoia.io Intakes and Playbook pages with write permissions
- Administrator access to the Google Cloud console

> #### Create a dedicated service account

Reviewer: ```suggestion
### Create a dedicated service account
```

To create a service account you have to :
 - Create a project
 - Turn on the APIs for the service account
 - Set up the OAuth consent screen
 - Create the service account

For more details in each steps please read this [Documentation](https://support.google.com/a/answer/7378726?hl=en)

> #### Create and download JSON keys (service account credentials)

Reviewer: ```suggestion
### Create and download JSON keys (service account credentials)
```

3. Click the email address of the service account that you want to create a key for
4. Click the **Keys** tab
5. Click the **Add key** drop-down menu, then select **Create new key**
6. Select **JSON** as the Key type and click **Create**

!!! Important
	Clicking **Create** downloads a service account key file. After you download the key file, you cannot download it again. You will need it on the following steps on Sekoia.io.

Find more information on the [official google documentation](https://cloud.google.com/iam/docs/keys-create-delete).

**Example of JSON key file**

```JSON
{
  "type": "service_account",
  "project_id": "PROJECT_ID",
  "private_key_id": "KEY_ID",
  "private_key": "-----BEGIN PRIVATE KEY-----\nPRIVATE_KEY\n-----END PRIVATE KEY-----\n",
  "client_email": "SERVICE_ACCOUNT_EMAIL",
  "client_id": "CLIENT_ID",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://accounts.google.com/o/oauth2/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/SERVICE_ACCOUNT_EMAIL"
}
```

## Sekoia.io configuration procedure

> #### Create your intake

Reviewer: ```suggestion
### Create your intake
```

1. Go to the [intake page](https://app.sekoia.io/operations/intakes) and create a new intake from the Google drive reports.
2. Copy the associated Intake key

> #### Pull the logs to collect them on Sekoia.io

Reviewer: ```suggestion
### Pull the logs to collect them on Sekoia.io
```

Go to the Sekoia.io [playbook page](https://app.sekoia.io/operations/playbooks), and follow these steps:

- Click on **+ PLAYBOOK** button to create a new one
- Select **Use a template**
- Search for `Google` keywork on the search bar and select the template named `Get activities from drive`

- Create a **Module configuration** using your service account credentials from your Google Cloud environment extracted on a JSON file. Name the module configuration as you wish

-  Create a **Trigger configuration** using:

	* `Intake key` created on the previous

- Click on the **Save** button
- **Activate the playbook** with the toggle button on the top right corner of the page

> #### Enjoy your events on the [Events page](https://app.sekoia.io/operations/events)

Reviewer: ```suggestion
### Enjoy your events on the [Events page](https://app.sekoia.io/operations/events)
```

uuid: bb11239e-3dc6-11ee-be56-0242ac120002
name: Google Drive Reports
type: intake

## Overview

Google Reports is a data reporting and analysis platform offered by Google, designed to provide insights and metrics about user activities and interactions within various Google services. It allows organizations to track and visualize user engagement, application usage, and other relevant data points, enabling informed decision-making and optimization of digital experiences. And Google Reports Drive is just for the Google Drive service.

{!_shared_content/operations_center/detection/generated/suggested_rules_bb11239e-3dc6-11ee-be56-0242ac120002_do_not_edit_manually.md!}

{!_shared_content/operations_center/integrations/generated/bb11239e-3dc6-11ee-be56-0242ac120002.md!}

## Google Cloud configuration procedure

### Prerequisites

- Google licence Enterprise standard or higher
- Access to Sekoia.io Intakes and Playbook pages with write permissions
- Administrator access to the Google Cloud console

### Create a dedicated service account

To create a service account you have to :

>  - Create a project
>  - Turn on the APIs for the service account
>  - Set up the OAuth consent screen
>  - Create the service account

Reviewer 2: Add another line between the previous sentence and the list for the layout
```suggestion

 - Create a project
 - Turn on the APIs for the service account
 - Set up the OAuth consent screen
 - Create the service account
```

4. Click the **Keys** tab
5. Click the **Add key** drop-down menu, then select **Create new key**
6. Select **JSON** as the Key type and click **Create**

!!! Important
	Clicking **Create** downloads a service account key file. After you download the key file, you cannot download it again. You will need it on the following steps on Sekoia.io.

Find more information on the [official google documentation](https://cloud.google.com/iam/docs/keys-create-delete).

**Example of JSON key file**

```JSON
{
  "type": "service_account",
  "project_id": "PROJECT_ID",
  "private_key_id": "KEY_ID",
  "private_key": "-----BEGIN PRIVATE KEY-----\nPRIVATE_KEY\n-----END PRIVATE KEY-----\n",
  "client_email": "SERVICE_ACCOUNT_EMAIL",
  "client_id": "CLIENT_ID",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://accounts.google.com/o/oauth2/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/SERVICE_ACCOUNT_EMAIL"
}
```

## Sekoia.io configuration procedure

### Create your intake

> 1. Go to the [intake page](https://app.sekoia.io/operations/intakes) and create a new intake from the Google drive reports.

Reviewer 2: I forgot that I changed the name of the format
```suggestion
1. Go to the [intake page](https://app.sekoia.io/operations/intakes) and create a new intake from the `Google Report`.
```
